### ISSUE TITLE
查看是否index错误，训练会出现out of range

### ISSUE THREAD

**Issue by qiwangzhixinxin** [yangjianxin1/Firefly#58]
在loss.py中：
      `# 模型前馈预测
        outputs = model(input_ids=input_ids, attention_mask=attention_mask, return_dict=True)
        logits = outputs["logits"] if isinstance(outputs, dict) else outputs[1]`
        在返回outputs[1]时会出现out of range， 是否index出现错误，建议修改成outputs[0]

**Comment by zhangxjohn** [comment]
修正成`outputs[0]`也不正确吧，通常情况下， logists是这么计算的：
```
hidden_states = outputs[0]
logits = self.lm_head(hidden_states)
```
而`self.lm_head =  nn.Linear(config.hidden_size, config.vocab_size, bias=False)`

这几天，在我运行的过程中，loss一直是波动的情况，我也在查找此问题。
所以，作者是否重新审视一下loss的计算是否正确？
@yangjianxin1

**Comment by zhangxjohn** [comment]
修改为outputs[0]不对吧，model的输出是
```
CausalLMOutputWithPast(
            loss=loss,
            logits=logits,
            past_key_values=outputs.past_key_values,
            hidden_states=outputs.hidden_states,
            attentions=outputs.attentions,
        )
```
修改为outputs[0]，返回了`loss`?

@yangjianxin1 看了您提交的修改，您重新考虑一下。

#58`查看是否index错误，训练会出现out of range`的问题是在`dataset`的问题，可以修改为：
```
utterances_ids = self.tokenizer(utterances, 
                                      add_special_tokens=False,
                                      truncation=True,
                                      max_length=self.max_seq_length).input_ids
```
该问题可以消除。@qiwangzhixinxin

**Comment by yangjianxin1** [comment]
感谢反馈，已经修复了👍🏻
